packages/web/src/components/map-layout.tsx

```diff
@@ -0,0 +1,85 @@
+import GoogleMapReact from "google-map-react";
+import React from "react";
+const markers = [];
+let labelIndex = 1;
+const deleteMarker = (marker, markers) => {
+    markers.splice(markers.indexOf(marker), 1);
+    marker.setMap(null);
+    marker = null;
+    markers.forEach((temp) => {
+        temp.setLabel(String(markers.indexOf(temp) + 1));
+    });
+    labelIndex = markers.length + 1;
+};
+const addMarker = (location, map, maps) => {
+    const marker = new maps.Marker({
```

Comment:
I see in the `google-map-react` package that you can also place custom React components on the map.  Which do you think we should use?  The native markers vs a custom component.  If we're going to use custom components then we'd store the markers in the component state instead of a global variable.

Also, should note that the routes shown on the map will also be added via the internal Google maps objects so they'd need to work well with the markers as well, custom or native.

Reply:
I wasn't figuring out how to use the React components to get the same functionality I wanted in terms of adding/deleting/managing markers and their props. There's probably a way to use this api. I'm just not familiar with React components enough or know if this api is expansive enough.